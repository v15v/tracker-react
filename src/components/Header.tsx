import React from "react";

interface Props {
    header: {
        title: string,
        subtitle: string
    }
}

const Header = ({header}: Props) => (
    <section className="hero is-info is-medium">
        <div className="hero-body">
            <div className="container is-widescreen">
                <p className="title">
                    {header ? header.title : ""}
                </p>
                <p className="subtitle">
                    {header ? header.subtitle : ""}
                </p>
            </div>
        </div>
    </section>
)

export default Header