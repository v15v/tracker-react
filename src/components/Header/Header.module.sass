.isMedium
  padding-bottom: 2rem