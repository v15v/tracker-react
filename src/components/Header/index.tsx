import React from "react";
import styles from "./Header.module.sass"
import {Link} from "react-router-dom";

interface Props {
    header: {
        title: string,
        subtitle: string
    }
}

const Header = ({header}: Props) => (
    <section className={`hero is-info ${styles.isMedium}`}>
        <div className="hero-body">
            <div className="container is-widescreen">
                <p className="title">
                    {header ? header.title : ""}
                </p>
                <p className="subtitle">
                    {header ? header.subtitle : ""}
                </p>
            </div>
        </div>
        <nav className="navbar" role="navigation" aria-label="main navigation">
            <div className="navbar-brand">
            </div>
            <div className="navbar-menu">
                <Link to={"/"} className="navbar-item">
                    Tracker
                </Link>
                <Link to={"/dashboard"} className="navbar-item">
                    Dashboard
                </Link>
                <Link to={"/calorie"} className="navbar-item">
                    CalorieDiary
                </Link>
            </div>
        </nav>
    </section>
)

export default Header