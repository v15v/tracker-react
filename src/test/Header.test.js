import {cleanup, render, screen} from '@testing-library/react';
import '@testing-library/jest-dom'
import Header from "../components/Header"

// Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
// unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

describe.skip('Header', () => {
    it("renders the Header with full data", () => {
        const header = {title: "Header title", subtitle: "Header subtitle"}
        render(<Header header={header}/>)

        expect(screen.getByText(/Header title/)).toBeInTheDocument()
        expect(screen.getByText(/Header subtitle/)).toBeInTheDocument()
    })
    it("renders the Header with other full data", () => {
        const header = {title: "Header title-1", subtitle: "Header subtitle-1"}
        render(<Header header={header}/>)

        expect(screen.getByText(/Header title-1/)).toBeInTheDocument()
        expect(screen.getByText(/Header subtitle-1/)).toBeInTheDocument()
    })
    it("renders the Header with title only", () => {
        const header = {title: "Header title"}
        render(<Header header={header}/>)

        expect(screen.getByText(/Header title/)).toBeInTheDocument()
    })
    it('Header snapshot', () => {
        const header = {title: "Header title", subtitle: "Header subtitle"}
        const view = render(<Header header={header}/>)

        expect(view).toMatchSnapshot()
    })
    it('Header without data snapshot', () => {
        const view = render(<Header/>)

        expect(view).toMatchSnapshot()
    })
})