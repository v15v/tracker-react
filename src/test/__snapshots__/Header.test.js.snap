// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="hero is-info isMedium"
      >
        <div
          class="hero-body"
        >
          <div
            class="container is-widescreen"
          >
            <p
              class="title"
            >
              Header title
            </p>
            <p
              class="subtitle"
            >
              Header subtitle
            </p>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="hero is-info isMedium"
    >
      <div
        class="hero-body"
      >
        <div
          class="container is-widescreen"
        >
          <p
            class="title"
          >
            Header title
          </p>
          <p
            class="subtitle"
          >
            Header subtitle
          </p>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header Header without data snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="hero is-info isMedium"
      >
        <div
          class="hero-body"
        >
          <div
            class="container is-widescreen"
          >
            <p
              class="title"
            />
            <p
              class="subtitle"
            />
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="hero is-info isMedium"
    >
      <div
        class="hero-body"
      >
        <div
          class="container is-widescreen"
        >
          <p
            class="title"
          />
          <p
            class="subtitle"
          />
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
